pkg/auth/authenticator/request/headerrequest/requestheader.go

```diff
@@ -32,22 +38,23 @@ func NewAuthenticator(providerName string, config *Config, mapper authapi.UserId
 }
 
 func (a *Authenticator) AuthenticateRequest(req *http.Request) (user.Info, bool, error) {
-	username := ""
-	for _, header := range a.config.UserNameHeaders {
-		header = strings.TrimSpace(header)
-		if len(header) == 0 {
-			continue
-		}
-		username = req.Header.Get(header)
-		if len(username) != 0 {
-			break
-		}
-	}
-	if len(username) == 0 {
+	id := headerValue(req.Header, a.config.IDHeaders)
+	if len(id) == 0 {
```

Comment:
Not that this is _incorrect_, but since this is now a function, wouldn't the logic be better if we had it return a boolean success or failure, rather than running a len() on the return? I'd think that would be less error-prone.